File: clients/dart/lib/model/inline_response2003_embedded_service_list.dart
part of openapi.api;

class InlineResponse2003EmbeddedServiceList {
  
  AnyType deletedAt = null;
  
  String networkId = null;
  
  String name = null;
  
  String updatedAt = null;
  
  InlineResponse2003EmbeddedModel model = null;
  
  AnyType zitiId = null;
  
  String ownerIdentityId = null;
  
  List<String> attributes = [];
  
  String modelType = null;
  
  String id = null;
  
  InlineResponse2003EmbeddedConfigIdByConfigTypeId configIdByConfigTypeId = null;
  
  String createdAt = null;
  
  String createdBy = null;
  
  AnyType deletedBy = null;
  
  bool encryptionRequired = null;
  
  AnyType authority = null;
  
  InlineResponse200Links links = null;
  InlineResponse2003EmbeddedServiceList();

  @override
  String toString() {
    return 'InlineResponse2003EmbeddedServiceList[deletedAt=$deletedAt, networkId=$networkId, name=$name, updatedAt=$updatedAt, model=$model, zitiId=$zitiId, ownerIdentityId=$ownerIdentityId, attributes=$attributes, modelType=$modelType, id=$id, configIdByConfigTypeId=$configIdByConfigTypeId, createdAt=$createdAt, createdBy=$createdBy, deletedBy=$deletedBy, encryptionRequired=$encryptionRequired, authority=$authority, links=$links, ]';
  }

  InlineResponse2003EmbeddedServiceList.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    deletedAt = json['deletedAt'];
    networkId = json['networkId'];
    name = json['name'];
    updatedAt = json['updatedAt'];
    model = (json['model'] == null) ?
      null :
      InlineResponse2003EmbeddedModel.fromJson(json['model']);
    zitiId = json['zitiId'];
    ownerIdentityId = json['ownerIdentityId'];
    attributes = (json['attributes'] == null) ?
      null :
      (json['attributes'] as List).cast<String>();
    modelType = json['modelType'];
    id = json['id'];
    configIdByConfigTypeId = (json['configIdByConfigTypeId'] == null) ?
      null :
      InlineResponse2003EmbeddedConfigIdByConfigTypeId.fromJson(json['configIdByConfigTypeId']);
    createdAt = json['createdAt'];
    createdBy = json['createdBy'];
    deletedBy = json['deletedBy'];
    encryptionRequired = json['encryptionRequired'];
    authority = json['authority'];
    links = (json['_links'] == null) ?
      null :
      InlineResponse200Links.fromJson(json['_links']);
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (deletedAt != null)
      json['deletedAt'] = deletedAt;
    if (networkId != null)
      json['networkId'] = networkId;
    if (name != null)
      json['name'] = name;
    if (updatedAt != null)
      json['updatedAt'] = updatedAt;
    if (model != null)
      json['model'] = model;
    if (zitiId != null)
      json['zitiId'] = zitiId;
    if (ownerIdentityId != null)
      json['ownerIdentityId'] = ownerIdentityId;
    if (attributes != null)
      json['attributes'] = attributes;
    if (modelType != null)
      json['modelType'] = modelType;
    if (id != null)
      json['id'] = id;
    if (configIdByConfigTypeId != null)
      json['configIdByConfigTypeId'] = configIdByConfigTypeId;
    if (createdAt != null)
      json['createdAt'] = createdAt;
    if (createdBy != null)
      json['createdBy'] = createdBy;
    if (deletedBy != null)
      json['deletedBy'] = deletedBy;
    if (encryptionRequired != null)
      json['encryptionRequired'] = encryptionRequired;
    if (authority != null)
      json['authority'] = authority;
    if (links != null)
      json['_links'] = links;
    return json;
  }

  static List<InlineResponse2003EmbeddedServiceList> listFromJson(List<dynamic> json) {
    return json == null ? List<InlineResponse2003EmbeddedServiceList>() : json.map((value) => InlineResponse2003EmbeddedServiceList.fromJson(value)).toList();
  }

  static Map<String, InlineResponse2003EmbeddedServiceList> mapFromJson(Map<String, dynamic> json) {
    var map = Map<String, InlineResponse2003EmbeddedServiceList>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = InlineResponse2003EmbeddedServiceList.fromJson(value));
    }
    return map;
  }

  // maps a json object with a list of InlineResponse2003EmbeddedServiceList-objects as value to a dart map
  static Map<String, List<InlineResponse2003EmbeddedServiceList>> mapListFromJson(Map<String, dynamic> json) {
    var map = Map<String, List<InlineResponse2003EmbeddedServiceList>>();
     if (json != null && json.isNotEmpty) {
       json.forEach((String key, dynamic value) {
         map[key] = InlineResponse2003EmbeddedServiceList.listFromJson(value);
       });
     }
     return map;
  }
}

File: clients/clojure/src/untitled_api/specs/inline_response_200_3_embedded_service_list.clj
(ns untitled-api.specs.inline-response-200-3-embedded-service-list
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [untitled-api.specs. :refer :all]
            [untitled-api.specs.inline-response-200-3-embedded-model :refer :all]
            [untitled-api.specs. :refer :all]
            [untitled-api.specs.inline-response-200-3-embedded-config-id-by-config-type-id :refer :all]
            [untitled-api.specs. :refer :all]
            [untitled-api.specs. :refer :all]
            [untitled-api.specs.inline-response-200-links :refer :all]
            )
  (:import (java.io File)))


(def inline-response-200-3-embedded-service-list-data
  {
   (ds/req :deletedAt) any-type-spec
   (ds/req :networkId) string?
   (ds/req :name) string?
   (ds/req :updatedAt) string?
   (ds/req :model) inline-response-200-3-embedded-model-spec
   (ds/req :zitiId) any-type-spec
   (ds/req :ownerIdentityId) string?
   (ds/req :attributes) (s/coll-of string?)
   (ds/req :modelType) string?
   (ds/req :id) string?
   (ds/req :configIdByConfigTypeId) inline-response-200-3-embedded-config-id-by-config-type-id-spec
   (ds/req :createdAt) string?
   (ds/req :createdBy) string?
   (ds/req :deletedBy) any-type-spec
   (ds/req :encryptionRequired) boolean?
   (ds/req :authority) any-type-spec
   (ds/req :_links) inline-response-200-links-spec
   })

(def inline-response-200-3-embedded-service-list-spec
  (ds/spec
    {:name ::inline-response-200-3-embedded-service-list
     :spec inline-response-200-3-embedded-service-list-data}))